scripts/validate-examples.sh

```diff
@@ -6,13 +6,28 @@ exampleDirs=$(find ./examples -mindepth 2 -maxdepth 2 -type d '!' -exec test -e
 examplesWithErrors=()
 hasError=false
 
+# Setup a local Terraform config file for setting up the dev_overrides for this provider.
+terraformrc=$(mktemp)
+cat << EOF > $terraformrc
+provider_installation {
+  dev_overrides {
+    "hashicorp/azurerm" = "$(go env GOPATH)/bin"
+  }
+  direct {}
+}
+EOF
+
 # first check each example validates via `terraform validate`..
 for d in $exampleDirs; do
   echo "Validating $d.."
   exampleHasErrors=false
+  # Though we are using the local built azurerm provider to validate example,
+  # we still need to call `terraform init` here as examples might contain other providers.
   terraform -chdir=$d init > /dev/null || exampleHasErrors=true
```

Comment:
do we not also need this for the init?

```suggestion
  TF_CLI_CONFIG_FILE=$terraformrc terraform -chdir=$d init > /dev/null || exampleHasErrors=true
```

Reply:
This is optional indeed. Arguably, without this we can further validate that we can initialize the provider from registry?

Reply:
I think this is needed to alleviate confusion, as otherwise the latest Provider binary will be downloaded (and added to the checksums) when it's not used?

Reply:
Sure, will do the change.